tasks/getters/getCastIntentTaskProps.js

```diff
@@ -0,0 +1,16 @@
+import { createStructuredSelector } from 'reselect'
+import getCurrentAgent from 'dogstack-agents/agents/getters/getCurrentAgent'
+import getProducts from '../../supply/getters/getProducts'
+import getResourceTypes from '../../resources/getters/getResourceTypes'
+import getPriceSpecs from '../../supply/getters/getPriceSpecsByProduct'
+import getRouterParams from '../../app/getters/getRouterParams'
+import getOrderIntents from '../../ordering/getters/getOrderIntents'
+
+export default createStructuredSelector({
+  currentAgent: getCurrentAgent,
+  products: getProducts,
```

Comment:
oh, also. this getter returns *all the products* that have been loaded. but in the respective query and render code, it's being treated as if it's the products for the given order. in this case we want a getter for the current order and a getter indexing the products by order (creating an object where keys are order ids and values are arrays of products with the corresponding order id). happy to do this.

Reply:
oops, products don't relate to an order directly, they relate to suppliers which relate to an order. soo, group products by supplier, key supplier by order, given order get supplier and then get products.

Reply:
we already have `getCurrentSupplierProducts`. :sweat_smile: